src/Microsoft.Azure.WebJobs.Host/InvokeFunctionFilterAttribute.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// An invocation filter that invokes job methods
+    /// </summary>
+    [CLSCompliant(false)]
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = true, AllowMultiple = true)]
+    public sealed class InvokeFunctionFilterAttribute : InvocationFilterAttribute
+    {
+        /// <summary>
+        /// When this attribute is used, take the name of the function to invoke
+        /// </summary>
+        /// <param name="executingFilter"></param>
+        /// <param name="executedFilter"></param>
+        public InvokeFunctionFilterAttribute(string executingFilter = null, string executedFilter = null)
+        {
+            ExecutingFilter = executingFilter;
+            ExecutedFilter = executedFilter;
+        }
+
+        /// <summary>
+        /// Executing filter
+        /// </summary>
+        public string ExecutingFilter { get; }
+
+        /// <summary>
+        /// Executed filter
+        /// </summary>
+        public string ExecutedFilter { get; }
+
+        /// <summary>
+        /// Call the requested job method before the main function call
+        /// </summary>
+        /// <returns></returns>
+        public override async Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            if (executingContext == null)
+            {
+                throw new ArgumentNullException("executingContext");
```

Comment:
nit: here and throughout, replace these string literal param names with use of the nameof operator, e.g. nameof(executingContext). That helps keep the code in sync in the face of renames